### DISCUSSION TITLE
How to get the current index and total runs in a multirun?

### DISCUSSION THREAD

**Discussion by kunaltyagi** [facebookresearch/hydra#2760] [Q&A]
If I start a run with `python main.py -m +x="range(0,3)"`, I wish to get that there are 3 runs in total, with the current run being `i` (in {1,2,3} or {0,1,2}: doesn't matter).

I don't see any documentation for this. Is this possible? If not, how can this be achieved?

**Comment by odelalleau** [comment]
It may be in `HydraConfig` (not 100% sure), you can inspect it with:

```python
from hydra.core.hydra_config import HydraConfig
from omegaconf import OmegaConf

hydra_cfg = HydraConfig.get()
print(OmegaConf.to_yaml(hydra_cfg))
```

**Reply by kunaltyagi** [reply]
What should I be looking for in there?

**Reply by odelalleau** [reply]
For the current index and total number of runs since that's what you asked for :p
(again I'm not sure it's there, it's just a guess)

**Reply by kunaltyagi** [reply]
`hydra.job.num` and `hydra.job.id` both give the current index, but not the total number of runs

**Reply by Jasha10** [reply]
@kunaltyagi I don't believe that the total number of runs is calculated ahead of time or stored in the hydra config.
For exmaple, take a look at the file `hydra/_internal/core_plugins/basic_sweeper.py` from the hydra source code. You will see that there is a loop of the form
```python
while not self.is_done():
    batch = self.get_job_batch()
    ...
    results = self.launcher.launch(batch, initial_job_idx=initial_job_idx)
    ...
    returns.append(results)
```
This `while` loop does not keep track of how many jobs have been run. You could potentially create a custom launcher or custom sweeper (based on the `basic_launcher` or the `basic_sweeper`) to calculate the total number of jobs that will be run before starting the loop.

**Reply by kunaltyagi** [reply]
I see that we have the length ready in https://github.com/facebookresearch/hydra/blob/809718cdcd64f9cd930d26dea69f2660a6ffa833/hydra/_internal/core_plugins/basic_sweeper.py#L158 (just need to use `len`)

It seems a potential PR to enable this in basic sweeper is possible.How do I supply that to hydra conf?

**Reply by kunaltyagi** [reply]
Any idea @Jasha10 ?

**Comment by lukaslobmaier** [comment]
A simple solution for hydra v1.2:

The value is stored in the hydra config:
https://github.com/facebookresearch/hydra/issues/1 Hydra config
config=hydra.core.hydra_config.HydraConfig.get()
https://github.com/facebookresearch/hydra/issues/2 Acces MultirunType
run_type=hydra.core.hydra_config.HydraConfig.get().mode
https://github.com/facebookresearch/hydra/issues/3 Total num of runs
total_num=hydra.core.hydra_config.HydraConfig.get().mode.value
https://github.com/facebookresearch/hydra/issues/4 Current num of job (only defined for multi runs)
current_num=hydra.core.hydra_config.HydraConfig.get().job.id

**Reply by MLNissenGonzalez** [reply]
hydra.core.hydra_config.HydraConfig.get().mode.value is either 1 (job) or 2 (multirun), see https://github.com/facebookresearch/hydra/blob/b785290504577b06d7131d87f85cfe121004dce2/hydra/types.py#L45

**Comment by MLNissenGonzalez** [comment]
Assuming you use the basic hydra sweeper, you can access the parameters you sweep over and compute the total number runs. Group is comma seperated string of options

```runtime_cfg = hydra.core.hydra_config.HydraConfig.get()
total_num = 1
params = runtime_cfg.sweeper.params
for group in params.values():
   options = group.split(",")
   total_num *= len(options)
